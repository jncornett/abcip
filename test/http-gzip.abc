# create a gzipped http response like this:

#$ gzip -c - < data.txt > data.gzip

#$ ls -l data.gzip
#-rw-r--r-- 1 russ russ 480 2011-10-16 10:32 data.gzip

# put the gzipped size in the content-length header.

c ( b=80; b.file="tmp/data.gzip" )

a ( syn )
b ( syn, ack )
a ( ack )

a ( ack, pay="GET / HTTP/1.0|d a d a|" )

b ( ack, pay=
    "HTTP/1.1 200 OK|d a|"
    "Content-Type: text/plain; charset=iso-8859-1|d a|"
    "Content-Encoding: gzip|d a|"
    "Content-Length: 480|d a d a|"
)

b ( ack, len=480 )
a ( ack )

b ( fin, ack )
a ( fin, ack )
b ( ack )
