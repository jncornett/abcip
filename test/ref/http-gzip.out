IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [S], cksum 0xb391 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0001 0000 0000 5002  .....P........P.
	0x0030:  2000 b391 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [S.], cksum 0xb37f (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0050 bdec 0000 0001 0000 0002 5012  ...P..........P.
	0x0030:  2000 b37f 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xb380 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 b380 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 58)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xd4ce (correct), seq 1:19, ack 1, win 8192, length 18
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  003a 0003 0000 4006 5ca8 0a01 0203 0a09  .:....@.\.......
	0x0020:  0807 bdec 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 d4ce 0000 4745 5420 2f20 4854 5450  ......GET./.HTTP
	0x0040:  2f31 2e30 0d0a 0d0a                      /1.0....
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 150)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0x04ca (correct), seq 1:111, ack 19, win 8192, length 110
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0096 0002 0000 4006 5c4d 0a09 0807 0a01  ......@.\M......
	0x0020:  0203 0050 bdec 0000 0002 0000 0014 5010  ...P..........P.
	0x0030:  2000 04ca 0000 4854 5450 2f31 2e31 2032  ......HTTP/1.1.2
	0x0040:  3030 204f 4b0d 0a43 6f6e 7465 6e74 2d54  00.OK..Content-T
	0x0050:  7970 653a 2074 6578 742f 706c 6169 6e3b  ype:.text/plain;
	0x0060:  2063 6861 7273 6574 3d69 736f 2d38 3835  .charset=iso-885
	0x0070:  392d 310d 0a43 6f6e 7465 6e74 2d45 6e63  9-1..Content-Enc
	0x0080:  6f64 696e 673a 2067 7a69 700d 0a43 6f6e  oding:.gzip..Con
	0x0090:  7465 6e74 2d4c 656e 6774 683a 2034 3830  tent-Length:.480
	0x00a0:  0d0a 0d0a                                ....
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 520)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0x93e6 (correct), seq 111:591, ack 19, win 8192, length 480
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0208 0003 0000 4006 5ada 0a09 0807 0a01  ......@.Z.......
	0x0020:  0203 0050 bdec 0000 0070 0000 0014 5010  ...P.....p....P.
	0x0030:  2000 93e6 0000 1f8b 0800 ceea 9a4e 0003  .............N..
	0x0040:  c553 4b6e dc30 0cdd eb14 f46c b2f1 f400  .SKn.0.....l....
	0x0050:  cda2 48a6 8ba0 2890 45e6 0294 4d5b 9a91  ..H...(.E...M[..
	0x0060:  cc40 9fb8 be7d 4979 d2cc 0dba 3104 928f  .@...}Iy....1...
	0x0070:  8f7c 8f7e 38af 98c1 261f 829f 7bc0 6584  .|.~8...&...{.e.
	0x0080:  e208 72f0 c56d 50f8 83b2 f9e9 4798 b744  ..r..mP.....G..D
	0x0090:  2d3b fb68 0381 5f5a dd8a 961e cd53 0810  -;.h.._Z.....S..
	0x00a0:  7dcc 1bac 2465 9ab0 9c78 5674 6f9e 6e3d  }...$e...xVto.n=
	0x00b0:  2347 8284 c565 e05a e624 d06f c61c 9e69  #G...e.Z.$.o...i
	0x00c0:  c51b e817 5a4b 69e5 e1da 43dc 20f3 d299  ....ZKi...C.....
	0x00d0:  b3c4 2fb8 6629 c002 d617 ea5b ed10 fe05  ../.f).....[....
	0x00e0:  072c 83eb cc7d a36a 2fd5 4a75 1af7 9db2  .,...}.j/.Ju....
	0x00f0:  ab4b 6b35 a51a 3dd7 0ccf 12a6 9497 863d  .Kk5..=........=
	0x0100:  18f3 2248 e62b 389f e183 d33b 06c8 2ba7  .."H.+8....;..+.
	0x0110:  5137 7552 fbdd fce6 6586 e2e3 4e11 71f9  Q7uR....e...N.q.
	0x0120:  a30b 4d4c a082 719d 5d39 1ecd 1b43 a25c  ..ML..q.]9...C.\
	0x0130:  68d4 a8dd 5aed b9c6 5223 9444 b4cb 9185  h...Z...R#.D....
	0x0140:  6a04 5c9d ff94 b2c1 450e cd8a 2112 abd3  j.\.....E...!...
	0x0150:  e4b3 fb4c 350e 45f5 6d8d 7ba5 56b1 0a68  ...L5.E.m.{.V..h
	0x0160:  2391 7582 2960 1492 937c 41da 4f53 f03a  #.u.)`...|A.OS.:
	0x0170:  b54b daa5 0d53 6a98 499c 6abd 94ce d624  .K...Sj.I.j....$
	0x0180:  8eee b42a 66a4 ce98 d745 755e b983 afd7  ...*f....Eu^....
	0x0190:  7e1c 7ba7 bb77 1be8 2699 0d38 0a31 2d05  ~.{..w..&..8.1-.
	0x01a0:  f2e2 872b a5a3 483c 5c3b d537 d054 9462  ...+..H<\;.7.T.b
	0x01b0:  24bc d9d2 46f7 25cb 7638 6a49 43ce 186a  $...F.%.v8jIC..j
	0x01c0:  7c77 3ab8 15ac 1e89 cee9 3097 2687 5c27  |w:.......0.&.\'
	0x01d0:  deee ef4b 871f e6a4 7614 d6cb c114 733b  ...K....v.....s;
	0x01e0:  214b 1855 45cb 5b67 5ec5 7db4 1775 7fc4  !K.UE.[g^.}..u..
	0x01f0:  ad83 1386 c0ec f687 040e 3ac1 e038 1595  ..........:..8..
	0x0200:  437d 975b b8f0 2603 3cfc e7df e42f 03e9  C}.[..&.<..../..
	0x0210:  42fb a703 0000                           B.....
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [.], cksum 0xb120 (correct), ack 591, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 5cb9 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0014 0000 0250 5010  .....P.......PP.
	0x0030:  2000 b120 0000                           ......
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [F.], cksum 0xb11f (correct), seq 591, ack 19, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 5cb9 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0050 bdec 0000 0250 0000 0014 5011  ...P.....P....P.
	0x0030:  2000 b11f 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.80: Flags [F.], cksum 0xb11e (correct), seq 19, ack 592, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 5cb8 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0050 0000 0014 0000 0251 5011  .....P.......QP.
	0x0030:  2000 b11e 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.80 > 10.1.2.3.48620: Flags [.], cksum 0xb11e (correct), ack 20, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 5cb8 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0050 bdec 0000 0251 0000 0015 5010  ...P.....Q....P.
	0x0030:  2000 b11e 0000                           ......
